controllers/toolchaincluster/healthchecker_test.go

```diff
@@ -0,0 +1,177 @@
+package toolchaincluster
+
+import (
+	"context"
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/cluster"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/h2non/gock.v1"
+	corev1 "k8s.io/api/core/v1"
+	kubeclientset "k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var logger = logf.Log.WithName("toolchaincluster_healthcheck")
+
+func TestClusterHealthChecks(t *testing.T) {
+
+	// given
+	defer gock.Off()
+	tcNs := "test-namespace"
+	gock.New("http://cluster.com").
+		Get("healthz").
+		Persist().
+		Reply(200).
+		BodyString("ok")
+	gock.New("http://unstable.com").
+		Get("healthz").
+		Persist().
+		Reply(200).
+		BodyString("unstable")
+	gock.New("http://not-found.com").
+		Get("healthz").
+		Persist().
+		Reply(404)
+
+	tests := map[string]struct {
+		tctype            string
+		apiendpoint       string
+		clusterconditions []toolchainv1alpha1.ToolchainClusterCondition
+		status            toolchainv1alpha1.ToolchainClusterStatus
+	}{
```

Comment:
I like this way you improved the unit test :+1: :1st_place_medal:

Reply:
+1